src/Neo/Ledger/SmartThrottler.cs

```diff
@@ -0,0 +1,185 @@
+        _lastBlockTimestamp = _lastResetTime.ToTimestampMS();
+        _averageFee = CalculateAverageFee(null);
+    }
+
+    /// <summary>
+    /// Determines whether a new transaction should be accepted
+    /// </summary>
+    /// <param name="tx">The transaction to be evaluated</param>
+    /// <returns>True if the transaction should be accepted, false otherwise</returns>
+    public bool ShouldAcceptTransaction(Transaction tx)
+    {
+        var now = TimeProvider.Current.UtcNow;
+        if (now - _lastResetTime >= TimeSpan.FromSeconds(1))
+        {
+            _transactionsThisSecond = 0;
+            _lastResetTime = now;
+            AdjustThrottling(null);
+            _averageFee = CalculateAverageFee(null);
+        }
+
+        // Check if we've hit the tx limit and it's not high priority
+        if ((_transactionsThisSecond >= _maxTransactionsPerSecond) && !IsHighPriorityTransaction(tx))
+            return false;
+
+        _transactionsThisSecond++;
+        return true;
+    }
+
+    /// <summary>
+    /// Updates the network state after a new block is added
+    /// </summary>
+    /// <param name="block">The newly added block</param>
+    public void UpdateNetworkState(Block block)
+    {
+        var currentTime = TimeProvider.Current.UtcNow.ToTimestampMS();
+        var blockTime = currentTime - _lastBlockTimestamp;
+
+        _recentBlockTimes.Enqueue(blockTime);
+        if (_recentBlockTimes.Count > BlockTimeWindowSize)
+            _recentBlockTimes.Dequeue();
+
+        _lastBlockTimestamp = currentTime;
+        _unconfirmedTxCount = _memoryPool.Count;
+        _averageFee = CalculateAverageFee(block);
+
+        AdjustThrottling(block);
+    }
+
+    /// <summary>
+    /// Adjusts throttling parameters based on current network conditions
+    /// </summary>
+    private void AdjustThrottling(Block? block)
+    {
+        var memoryPoolUtilization = (double)_memoryPool.Count / _system.Settings.MemoryPoolMaxTransactions;
+        var networkLoad = EstimateNetworkLoad(block);
+
+        _maxTransactionsPerSecond = CalculateOptimalTps(memoryPoolUtilization, networkLoad, block);
+    }
+
+    /// <summary>
+    /// Estimates current network load
+    /// </summary>
+    /// <returns>An integer between 0 and 100 representing the estimated network load</returns>
+    private int EstimateNetworkLoad(Block? block)
+    {
+        var load = 0;
+
+        // 1. Memory pool utilization (30% weight)
+        var memPoolUtilization = (double)_memoryPool.Count / _system.Settings.MemoryPoolMaxTransactions;
+        load += (int)(memPoolUtilization * 30);
+
+        // 2. Recent block times (30% weight)
+        if (_recentBlockTimes.Count > 0)
+        {
+            var avgBlockTime = _recentBlockTimes.Average(t => (double)t);
+            var blockTimeRatio = avgBlockTime / _system.Settings.MillisecondsPerBlock;
+            load += (int)(Math.Min(blockTimeRatio, 2) * 30); // Cap at 60 points
```

Comment:
I am not sure.

avgBlockTime should usually not be lower than " _system.Settings.MillisecondsPerBlock".
So, usually blockTimeRatio is close to "1".

So, there is an average load from 30 - 60

Reply:
its either 0 or 30 actually.

Reply:
How?

`(int)(Math.Min(blockTimeRatio, 2) * 30);`

BlockTimeRatio will never/RARELY be less than 1.
So the range is 30 - 60 in this formula